clang-10 unit_tests compilation error in lambda capture

When compiling the unit tests using the recently released clang-10 the following compile-time error is encountered:
```
/tmp/hipSYCL/tests/unit_tests.cpp:636:27: error: variable 'd' cannot be implicitly captured in a lambda with no capture-default specified
        [=](const s::item<d> item) {
                          ^
/tmp/hipSYCL/tests/unit_tests.cpp:636:28: note: while checking a default template argument used here
        [=](const s::item<d> item) {
                     ~~~~~~^
/tmp/hipSYCL/tests/unit_tests.cpp:615:18: note: 'd' declared here
  constexpr auto d = _dimensions::value;
```
LLVM has been built from source, from the branch [release/10.x](https://github.com/llvm/llvm-project/tree/release/10.x). The used binaries are available in our repo at http://repo.urz.uni-heidelberg.de/sycl/test_llvm10/

It is curious that compiling using clang-9 works fine.

--- a/tests/unit_tests.cpp
+++ b/tests/unit_tests.cpp
@@ -612,7 +612,8 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(id_api, _dimensions, test_dimensions::type) {
 
 BOOST_AUTO_TEST_CASE_TEMPLATE(item_api, _dimensions, test_dimensions::type) {
   namespace s = cl::sycl;
-  constexpr auto d = _dimensions::value;
+  // Specify type explicitly to workaround Clang bug #45538
+  constexpr int d = _dimensions::value;
 
   const auto test_range = make_test_value<s::range, d>(
     { 5 }, { 5, 7 }, { 5, 7, 11 });
@@ -851,7 +852,8 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(item_api, _dimensions, test_dimensions::type) {
 BOOST_AUTO_TEST_CASE_TEMPLATE(explicit_buffer_copy_host_ptr, _dimensions,
   explicit_copy_test_dimensions::type) {
   namespace s = cl::sycl;
-  constexpr auto d = _dimensions::value;
+  // Specify type explicitly to workaround Clang bug #45538
+  constexpr int d = _dimensions::value;
 
   const auto buf_size = make_test_value<s::range, d>(
     { 64 }, { 64, 96 }, { 64, 96, 48 });

